lib/src/extensions/channel.dart

```diff
@@ -24,6 +24,91 @@ extension GuildChannelExtensions on GuildChannel {
   Future<Permissions> computePermissionsFor(PartialMember member) async => await computePermissions(this, await member.get());
 }
 
+/// Extensions on all [GuildChannel] types
+extension GuildChannelsExtension<T extends GuildChannel> on T {
+  CreateBuilder<T> toBuilder() {
```

Comment:
I would prefer switching based on type rather than a field, e.g:
```dart
return switch (this) {
  TextGuildChannel channel => GuildChannelBuilder.text(...),
  VoiceChannel channel => GuildChannelBuilder.voice(...),
  // etc.
}
```

Reply:
I See, as often I thought way too complicated...